core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1101,6 +1109,46 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
     return n;
   }
 
+  //template<class FunctorType>
+  struct ThrustFunctorWrapper {
+    const FunctorType& f;
+    KOKKOS_FUNCTION
+    ThrustFunctorWrapper(const FunctorType& op):f(op){};
+    KOKKOS_FUNCTION
+    value_type operator() (index_type i) const {
```

Comment:
you need two overloads here: one with tag type void and one with a non void tag type look in the original place for calling m_functor.